src/lib/FiatApi.js

```diff
@@ -11,109 +11,138 @@ class FiatApi {
+        CHF: /** @type {'chf'} */ ('chf'), // Swiss Franc
+        CLP: /** @type {'clp'} */ ('clp'), // Chilean Peso
+        CNY: /** @type {'cny'} */ ('cny'), // Chinese Yuan
+        CZK: /** @type {'czk'} */ ('czk'), // Czech Koruna
+        DKK: /** @type {'dkk'} */ ('dkk'), // Danish Krone
+        EUR: /** @type {'eur'} */ ('eur'), // Euro
+        GBP: /** @type {'gbp'} */ ('gbp'), // British Pound
+        GEL: /** @type {'gel'} */ ('gel'), // Georgian Lari
+        HKD: /** @type {'hkd'} */ ('hkd'), // Hong Kong Dollar
+        HUF: /** @type {'huf'} */ ('huf'), // Hungarian Forint
+        IDR: /** @type {'idr'} */ ('idr'), // Indonesian Rupiah
+        ILS: /** @type {'ils'} */ ('ils'), // Israeli New Shekel
+        INR: /** @type {'inr'} */ ('inr'), // Indian Rupee
+        JPY: /** @type {'jpy'} */ ('jpy'), // Japanese Yen
+        KRW: /** @type {'krw'} */ ('krw'), // South Korean Won
+        KWD: /** @type {'kwd'} */ ('kwd'), // Kuwaiti Dinar
+        LKR: /** @type {'lkr'} */ ('lkr'), // Sri Lankan Rupee
+        MMK: /** @type {'mmk'} */ ('mmk'), // Burmese Kyat
+        MXN: /** @type {'mxn'} */ ('mxn'), // Mexican Peso
+        MYR: /** @type {'myr'} */ ('myr'), // Malaysian Ringgit
+        NGN: /** @type {'ngn'} */ ('ngn'), // Nigerian Naira
+        NOK: /** @type {'nok'} */ ('nok'), // Norwegian Krone
+        NZD: /** @type {'nzd'} */ ('nzd'), // New Zealand Dollar
+        PHP: /** @type {'php'} */ ('php'), // Philippine Peso
+        PKR: /** @type {'pkr'} */ ('pkr'), // Pakistani Rupee
+        PLN: /** @type {'pln'} */ ('pln'), // Poland Złoty
+        RUB: /** @type {'rub'} */ ('rub'), // Russian Ruble
+        SAR: /** @type {'sar'} */ ('sar'), // Saudi Riyal
+        SEK: /** @type {'sek'} */ ('sek'), // Swedish Krona
+        SGD: /** @type {'sgd'} */ ('sgd'), // Singapore Dollar
+        THB: /** @type {'thb'} */ ('thb'), // Thai Baht
+        TRY: /** @type {'try'} */ ('try'), // Turkish Lira
+        TWD: /** @type {'twd'} */ ('twd'), // New Taiwan Dollar
+        UAH: /** @type {'uah'} */ ('uah'), // Ukrainian Hryvnia
+        USD: /** @type {'usd'} */ ('usd'), // United States Dollar
+        // VEF: /** @type {'vef'} */ ('vef'), // Discontinued Venezuelan Bolívar Fuerte replaced by VES. Rates are off.
+        VND: /** @type {'vnd'} */ ('vnd'), // Vietnamese Đồng
+        ZAR: /** @type {'zar'} */ ('zar'), // South African Rand
+    };
+
+    /**
+     * @readonly
+     * Coingecko api url. Note that the origin must be whitelisted in the csp.
+     */
+    FiatApi.API_URL = 'https://api.coingecko.com/api/v3';
+
+    /**
+     * @readonly
+     * Crypto currency tickers mapped to coingecko coin ids.
+     */
+    FiatApi.COINGECKO_COIN_IDS = {
+        [FiatApi.SupportedCryptoCurrency.NIM]: 'nimiq-2',
+        [FiatApi.SupportedCryptoCurrency.BTC]: 'bitcoin',
+    };
+} else {
+    throw new Error(`FiatApi.Provider ${FiatApi.Provider} not supported yet.`);
+}
```

Comment:
Can we also convert this `if-else` statement into a switch statement without a `default` case, so TS catches missing providers?